aten/src/ATen/TensorIterator.cpp

```diff
@@ -946,6 +946,8 @@ void TensorIteratorBase::build_ternary_op(
     const TensorBase& out, const TensorBase& a,
     const TensorBase& b, const TensorBase& c) {
   build(TensorIteratorConfig()
+      .promote_inputs_to_common_dtype(true)
```

Comment:
What other ternary operations are affected by this change and do they require additional testing?